jwst/tweakreg/tweakreg_step.py

```diff
@@ -170,17 +170,35 @@ def process(self, input):
             yoffset=self.yoffset
         )
 
-        align_wcs(
-            imcats,
-            refcat=None,
-            enforce_user_order=self.enforce_user_order,
-            expand_refcat=self.expand_refcat,
-            minobj=self.minobj,
-            match=tpmatch,
-            fitgeom=self.fitgeometry,
-            nclip=self.nclip,
-            sigma=(self.sigma, 'rmse')
-        )
+        try:
+            align_wcs(
+                imcats,
+                refcat=None,
+                enforce_user_order=self.enforce_user_order,
+                expand_refcat=self.expand_refcat,
+                minobj=self.minobj,
+                match=tpmatch,
+                fitgeom=self.fitgeometry,
+                nclip=self.nclip,
+                sigma=(self.sigma, 'rmse')
+            )
+
+        except ValueError as e:
+            msg = e.args[0]
+            if (msg == "Too few input images (or groups of images) with "
+                "non-empty catalogs."):
+                # we need at least two exposures to perform image alignment
+                self.log.warning(msg)
+                self.log.warning("At least two exposures are required for "
```

Comment:
Too late now, but in hind sight I think this warning message could've been a bit more explicit. The problem condition that's being encountered is not the fact that there aren't enough images, it's the fact that there aren't enough images **WITH DETECTED SOURCES**. I think we'll get confused users saying "I have several input images, so why is it telling me I need at least two?".

Reply:
Is the message given above, "Too few input images (or groups of images) with non-empty catalogs" not clear enough?  Or is the term "detected sources" preferable?

Reply:
If the message referred to in lines 188-189 is also seen by the user, then I think we're OK as-is. I had initially assumed that that message was being trapped and then only the one in lines 192-193 was being written out for the user.